File: week_8/day4/fizzbuzz-tdd/lib/fizz_buzz.rb
class FizzBuzz
  def run(n)
    return 'fizzbuzz' if (n % 15).zero?
    return 'buzz' if (n % 5).zero?
    return 'fizz' if (n % 3).zero?
    n
  end
end

fb = FizzBuzz.new
(1..100).each do |n|
  puts fb.run(n)
end

File: week_8/day4/fizzbuzz-tdd/lib/fizz_buzz.js
var FizzBuzz = {
    run: function (n) {
        if (this.isZero(n % 15)) { return 'fizzbuzz'; }
        if (this.isZero(n %  3)) { return 'fizz';     }
        if (this.isZero(n %  5)) { return 'buzz';     }

        return n;
    },

    isZero: function (pred) {
        return pred === 0;
    }
};

module.exports = FizzBuzz;
